core/src/main/java/org/mule/runtime/core/api/profiling/notification/ProcessingStrategyProfilingEventType.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.core.api.profiling.notification;
+
+import org.mule.api.annotation.Experimental;
+import org.mule.runtime.api.profiling.type.ProfilingEventType;
+import org.mule.runtime.core.api.profiling.consumer.context.ProcessingStrategyProfilingEventContext;
+
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A {@link ProfilingEventType} for processing strategies.
+ *
+ * @since 4.4
+ */
+@Experimental
+public class ProcessingStrategyProfilingEventType implements ProfilingEventType<ProcessingStrategyProfilingEventContext> {
+
+  private final String profilingEventTypeIdentifier;
+  private final String profilingEventTypeNamespace;
+
+  // TODO Can this be refactored and become a generic ProfilingEventType generator?
```

Comment:
I've tried to create a ProfilingEventType generator but this will result in returning a generic profiling event type. So in that case for compile-time type verification a cast will be needed or no compile time type verification for the profiling event context will be done. 
Not sure if this is appropriate. Create a ticket for technical debt for review.